### PR TITLE
All workitem services / models migrated to json api

### LINKED ISSUE #442: Update the WorkItem service / model to use the JSON API

### PR DESCRIPTION
**Pull request by sanbornsen**
- All services, components and templates are updated as per the new format.
- Updated the inmemory data as per the new data format.
- Updated the data in spec files for unit test as per the new format.
- fixes #442

Update Core to match API migration - https://github.com/aslakknutsen/almighty-core/tree/workitem_jsonapi

### CODE REVIEW

File: src/app/board/board.component.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { Component, OnInit } from '@angular/core';
 import { Router }            from '@angular/router';
 import { DragulaService } from 'ng2-dragula/ng2-dragula';
 
-import { WorkItem } from '../work-item/work-item';
+import { WorkItem } from './../models/work-item';
```
Comment (joshuawilson) [inline src/app/board/board.component.ts]: This does not require the extra "./" in front.

**Comment by sanbornsen** [comment]
The corresponding core PR - https://github.com/almighty/almighty-core/pull/559

File: src/app/in-memory-data.service.ts [hunk 2]
```diff
@@ -2,15 +2,58 @@ import { InMemoryDbService } from 'angular-in-memory-web-api';
...
+          'baseType': {
+            'data': {
+              'id': 'system.userstory',
+              'type': 'workitemtypes'
+            }
+          }
+        },
+        'type': 'workitems'
+      };
+      // return {'fields': {'system.assignee': 'someUser' + n,
```
Comment (joshuawilson) [inline src/app/in-memory-data.service.ts]: do we need to keep this commented code?

Comment (sanbornsen) [inline src/app/in-memory-data.service.ts]: Nope, I'll remove them.

File: src/app/work-item/work-item-list/work-item-detail/work-item-detail.component.spec.ts [hunk 3]
```diff
@@ -293,7 +307,7 @@ describe('Detailed view and edit a selected work item - ', () => {
     comp.loggedIn = fakeAuthService.isLoggedIn();
```
Comment (joshuawilson) [inline src/app/work-item/work-item-list/work-item-detail/work-item-detail.component.spec.ts]: The comp.workItem is showing a type error.
TS2322:Type 'WorkItem' is not assignable to type 'WorkItem'.

Comment (joshuawilson) [inline src/app/work-item/work-item-list/work-item-detail/work-item-detail.component.spec.ts]: The fix is in work-item-detail.component.ts and add "public" to workItem: WorkItem;

Comment (joshuawilson) [inline src/app/work-item/work-item-list/work-item-detail/work-item-detail.component.spec.ts]: ugh, so I tried the change and then undid the fix locally and I don't see the Type error anymore. Please at least take a look at it.

File: src/app/work-item/work-item-list/work-item-list.component.html [hunk 4]
```diff
@@ -20,7 +20,7 @@
 <div id="workItemList_Error" class="error" *ngIf="error">{{error}}</div>
 
 <div id="workItemList_quickAdd" class="pull-left">
-    <div *ngIf="loggedIn" class="pull-left supress-hover" (click)="addWorkItem()">
+    <div *ngIf="loggedIn" class="pull-left supress-hover">
```
Comment (joshuawilson) [inline src/app/work-item/work-item-list/work-item-list.component.html]: why did you remove the click event?

Comment (sanbornsen) [inline src/app/work-item/work-item-list/work-item-list.component.html]: We are not using this addWorkItem function any more. I don't know when it was removed but it was throwing an error. In fact the error is currently on production as well. We don't need this event anyway, so removed it.

File: src/app/work-item/work-item-quick-add/work-item-quick-add.component.spec.ts [hunk 5]
```diff
@@ -78,7 +87,7 @@ describe('Quick add work item component - ', () => {
     fixture.detectChanges();
     el = fixture.debugElement.query(By.css('.workItemQuickAdd_Add'));
     fixture.detectChanges();
-    comp.workItem.fields['system.title'] = '  ';
+    comp.workItem.attributes['system.title'] = '  ';
```
Comment (sanbornsen) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.spec.ts]: Are you seeing this on you IDE or console?

Comment (joshuawilson) [inline src/app/work-item/work-item-quick-add/work-item-quick-add.component.spec.ts]: I can see the Type errors in my IDE.
